blocks/library/categories/data.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns a Promise with the categories or an error on failure.
```

Comment:
Technically it returns a [`jqXHR`](http://api.jquery.com/jQuery.ajax/#jqXHR) which implements the [`jQuery.Deffered`](https://api.jquery.com/jquery.deferred/) interface (i.e. `.done`, not `.then`).

http://backbonejs.org/#Collection-fetch